src/store/indexeddb-local-backend.js

```diff
@@ -239,42 +223,39 @@ LocalIndexedDBStoreBackend.prototype = {
     },
 
     /**
-     * Persist a list of User objects. Users with the same 'userId' will be
-     * replaced.
-     * @param {User[]} users An array of users
+     * Persist a list of [user id, presence event] they are for.
+     * Users with the same 'userId' will be replaced.
+     * Prfesence events should be the event in its raw form (not the Event
+     * object)
+     * @param {Object[]} tuples An array of [userid, event] tuples
```

Comment:
> Tuples will be a little smaller

Source? Heapdumping an array and an object with the same data (and `userId` and `event` keys actually shows that:
```
Object form
  shallow size = 40 bytes
  retained size = 40 bytes

Array form
  shallow size = 32 bytes
  retained size = 64 bytes
```

Tested via:
```
window.ALPHABAT = {}
window.ALPHABAT.foo = ["alpha","bet"]
window.ALPHABAT.bar = {userId:"alpha",event:"bet"}
```

If `postMessage` sends over actual JSON then sure, array form will indeed be slightly smaller. I don't like how we've prematurely optimised for that at the sake of clarity, but as you say, it's an internal interface and I don't care enough to make you change this.